examples/chip-tool/commands/common/CHIPCommand.cpp

```diff
@@ -78,6 +79,12 @@ CHIP_ERROR CHIPCommand::Run()
 
     Shutdown();
 
+    // Commissioners may have active lookups in case of timeouts shut down all active
+    // resolves before destroying commissioners.
+    //
+    // safe to call since chip thrad and event queue are stopped, preventing any thread races.
+    chip::AddressResolve::Resolver::Instance().Shutdown();
```

Comment:
`Dnssd::Resolver::Instance().Init` happens in `CHIPDeviceControllerFactory`. Shouldn't Shutdown happen there too?

Reply:
Right, I think this will head towards the path that Boris is proposing:
 - current problem occurs because Controllers are shut down while a lookup is active (and that references them as callbacks) even though Resolver is still theoretically active
 - shutting down things in the controller factory is too late in that case. I will add it there, however I still need to be able to remove/shutdown controllers before the resolver is shut down globally.
 
 Will update this PR with that logic.

Reply:
Sounds like there has to be a distinction between 'shutdown just active lookups associated with a particular controller/fabric' vs. 'shutdown all lookups'.

On the former, how exactly are we going to shutdown look-ups if there are multiple controllers active for a given fabric?

Reply:
I believe shutdown on a global object only exists as a "application is shutting down". I do not want to make address resolver be aware of fabrics.

Reply:
Code updated to have lookup cancellation. I made the device proxy cancel active lookups if it is being destroyed.